azure/application-gateway/advanced/outputs.tf

```
@@ -31,4 +31,14 @@ output "frontend_ip_configuration" {
 output "identity_id" {
   description = "Identity of the AppGW if KV is used."
   value       = azurerm_user_assigned_identity.main_gateway[0].id
+}
+
+output "private_ip" {
+  description = "Private IP Address"
+  value = var.private_ip
+}
+
+output "public_ip" {
+  description = "Public IP Address"
+  value = azurerm_public_ip.main[0].ip_address
```

If not using the public IP, use one() to omit it if it's not found

```suggestion
  value = one(azurerm_public_ip.main[*].ip_address)
```